cardano-cli/test/Test/Golden/Shelley/Governance/VerifyPoll.hs

```diff
@@ -1,64 +1,96 @@
+{-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE OverloadedStrings #-}
 
 module Test.Golden.Shelley.Governance.VerifyPoll
-  ( golden_shelleyGovernanceVerifyPollVrf
-  , golden_shelleyGovernanceVerifyPollVrfTempered
-  , golden_shelleyGovernanceVerifyPollCold
-  , golden_shelleyGovernanceVerifyPollColdTempered
+  ( golden_shelleyGovernanceVerifyPoll
+  , golden_shelleyGovernanceVerifyPollMismatch
+  , golden_shelleyGovernanceVerifyPollNoAnswer
+  , golden_shelleyGovernanceVerifyPollMalformedAnswer
+  , golden_shelleyGovernanceVerifyPollInvalidAnswer
   ) where
 
-import           Control.Monad (void)
+import           Control.Monad.IO.Class (liftIO)
 import           Hedgehog (Property)
 import           Test.OptParse
 
+import           Cardano.Api
+import           Cardano.CLI.Shelley.Key
+                   (VerificationKeyOrFile (..),
+                   readVerificationKeyOrTextEnvFile)
+
 import qualified Hedgehog as H
+import qualified Data.ByteString.Char8 as BSC
 
 {- HLINT ignore "Use camelCase" -}
 
-golden_shelleyGovernanceVerifyPollVrf :: Property
-golden_shelleyGovernanceVerifyPollVrf = propertyOnce $ do
-  pollFile <- noteInputFile "test/data/golden/shelley/governance/poll.json"
-  metadataFile <- noteInputFile "test/data/golden/shelley/governance/answer-vrf.json"
+golden_shelleyGovernanceVerifyPoll :: Property
+golden_shelleyGovernanceVerifyPoll = propertyOnce $ do
+  pollFile <- noteInputFile "test/data/golden/shelley/governance/polls/basic.json"
+  txFile <- noteInputFile "test/data/golden/shelley/governance/verify/valid"
+  vkFile <- VerificationKeyFilePath . File <$>
+    noteInputFile "test/data/golden/shelley/governance/cold.vk"
 
-  void $ execCardanoCLI
+  stdout <- BSC.pack <$> execCardanoCLI
     [ "governance", "verify-poll"
     , "--poll-file", pollFile
-    , "--metadata-file", metadataFile
+    , "--tx-file", txFile
     ]
 
-golden_shelleyGovernanceVerifyPollCold :: Property
-golden_shelleyGovernanceVerifyPollCold = propertyOnce $ do
-  pollFile <- noteInputFile "test/data/golden/shelley/governance/poll.json"
-  metadataFile <- noteInputFile "test/data/golden/shelley/governance/answer-cold.json"
+  liftIO (readVerificationKeyOrTextEnvFile AsStakePoolKey vkFile) >>= \case
+    Left{} ->
+      H.failure
```

Comment:
You should render the failure with `failWith Nothing . Text.unpack $ displayError e.`